lib/extensions.js

```diff
@@ -40,6 +40,34 @@ module.exports = function(realmConstructor) {
     setConstructorOnPrototype(realmConstructor.Results);
     setConstructorOnPrototype(realmConstructor.Object);
 
+    //Add async open API
+    Object.defineProperties(realmConstructor, getOwnPropertyDescriptors({
+        open(config) {
+            return new Promise((resolve, reject) => {
+                const realm = new realmConstructor(config);
```

Comment:
@fealebenpae 
/cc @kristiandupont 

While I am working on using a schema-less Realm as the cocoa approach, isn't their approach achievable by doing 

```JavaScript
open(config) {
            return new Promise((resolve, reject) => {
                const realm = new realmConstructor();
                realm.wait((error) => {
                    if (error) {
                        reject(new Error("Cannot asynchronously open synced Realm, because the associated session previously experienced a fatal error"));
                    }

                    const syncedRealm = new realmConstructor(config);
                    resolve(syncedRealm);
                });
            });
        },
```

It boils down to the question what var realm = new Realm() is doing. Isn't this creating a schema-less local realm. By looking at the code it seems it does exactly that.

The end result of my query to you is that this change can help keep the current proposed api Realm.open. It's the internal implementation that will change a bit.

What do you think?